tests/test_input.py

```diff
@@ -432,54 +434,78 @@ def test_output_galaxy_properties(mock_galaxy):
     assert isinstance(g.paramcirc[2], u.Quantity)
 
 
-@pytest.mark.xfail
-def test_energy_method(disc_particles_all, halo_particles):
+def test_energy_method(mock_galaxy):
     """Test energy method."""
-    (mass_s, pos_s, vel_s, mass_g, pos_g, vel_g) = disc_particles_all
+    g = mock_galaxy
 
-    mass_dm, pos_dm, vel_dm = halo_particles
+    E_tot_dm, E_tot_s, E_tot_g = g.energy
 
-    g = galaxychop.Galaxy(
-        x_s=pos_s[:, 0] * u.kpc,
-        y_s=pos_s[:, 1] * u.kpc,
-        z_s=pos_s[:, 2] * u.kpc,
-        vx_s=vel_s[:, 0] * (u.km / u.s),
-        vy_s=vel_s[:, 1] * (u.km / u.s),
-        vz_s=vel_s[:, 2] * (u.km / u.s),
-        m_s=mass_s * u.M_sun,
-        x_dm=pos_dm[:, 0] * u.kpc,
-        y_dm=pos_dm[:, 1] * u.kpc,
-        z_dm=pos_dm[:, 2] * u.kpc,
-        vx_dm=vel_dm[:, 0] * (u.km / u.s),
-        vy_dm=vel_dm[:, 1] * (u.km / u.s),
-        vz_dm=vel_dm[:, 2] * (u.km / u.s),
-        m_dm=mass_dm * u.M_sun,
-        x_g=pos_g[:, 0] * u.kpc,
-        y_g=pos_g[:, 1] * u.kpc,
-        z_g=pos_g[:, 2] * u.kpc,
-        vx_g=vel_g[:, 0] * (u.km / u.s),
-        vy_g=vel_g[:, 1] * (u.km / u.s),
-        vz_g=vel_g[:, 2] * (u.km / u.s),
-        m_g=mass_g * u.M_sun,
-    )
+    k_s = 0.5 * (g.vx_s.value ** 2 + g.vy_s.value ** 2 + g.vz_s.value ** 2)
+    k_dm = 0.5 * (g.vx_dm.value ** 2 + g.vy_dm.value ** 2 + g.vz_dm.value ** 2)
+    k_g = 0.5 * (g.vx_g.value ** 2 + g.vy_g.value ** 2 + g.vz_g.value ** 2)
 
-    E_tot_dm, E_tot_s, E_tot_g = g.energy
+    x = np.hstack((g.x_s.value, g.x_dm.value, g.x_g.value))
```

Comment:
Here it is preferable to use the array accessor of uttr to avoid incorrect units. (e.g. `self.arr_.x_s`)

Reply:
Sorry, it is `g.arr_.x_s` not `self` because you need to call the object.